testing/README.md

```diff
@@ -0,0 +1,355 @@
+
+  ```js
+  // bad
+  expect(2 === 1).to.be.false;
+  (2 === 1).should.be.false;
+
+  // good
+  assert.isFalse(2 === 1);
+  ```
+
+- Use the Mocha-provided [TDD interface](https://mochajs.org/#tdd), which includes the `suite`, `test`, `setup`, and `teardown` functions.
+
+  ```js
+  // bad
+  describe('myFunction', () => {
+    beforeEach(() => {
+      doSomeSetup();
+    });
+
+    it('is a function', () => {
+      assert.isFunction(myFunction);
+    });
+  });
+
+  // good
+  suite('myFunction', () => {
+    setup(() => {
+      doSomeSetup();
+    });
+
+    test('it is a function', () => {
+      assert.isFunction(myFunction);
+    });
+  });
+  ```
+
+- Do not attach variables under test to the Mocha test context (`this`). Instead, use variables scoped to the nearest `suite`. If using ESNext, always use arrow functions to ensure that you will not be able to rely on context.
+
+  > Why? Using a shared context makes it too easy for tests to "leak" (have some impact on other tests), especially in other test files. It also makes it harder to track down what should be in scope at any given time.
+
+  ```js
+  // bad
+  suite('MyComponent', function() {
+    setup(function() {
+      this.subject = new MyComponent();
+      this.doSomethingWithComponent = function() {
+        return this.subject.doSomething();
+      }
+    });
+
+    test(function() {
+      assert.isFalse(this.doSomethingWithComponent());
+    });
+  });
+
+  // good
+  suite('MyComponent', () => {
+    let subject;
+    function doSomethingWithComponent() {
+      return subject.doSomething();
+    }
+
+    setup(() => {
+      subject = new MyComponent();
+    });
+
+    test(() => {
+      assert.isFalse(doSomethingWithComponent());
+    });
+  });
+  ```
+
+- Move any common setup to the `setup` hook of your suite.
+
+  > Why? Setup in the test itself draws attention from whatever is being tested.
+
+  ```js
+  suite('MyComponent', () => {
+    // bad
+    suite('#myBadFunc', () => {
+      test('it does something', () => {
+        const subject = new MyComponent();
+        sinon.stub(subject, 'myBadFunc').returns(true);
+        assert.true(subject.doSomething())
+      });
+
+      test('it does something else', () => {
+        const subject = new MyComponent();
+        sinon.stub(subject, 'myBadFunc').returns(true);
+        assert.false(subject.doSomethingElse())
+      });
+    });
+
+    // good
+    suite('#myBadFunc', () => {
+      let subject;
+
+      setup(() => {
+        subject = new MyComponent();
+        sinon.stub(subject, 'myBadFunc').returns(true);
+      });
+
+      test('it does something', () => {
+        assert.true(subject.doSomething())
+      });
+
+      test('it does something else', () => {
+        assert.false(subject.doSomethingElse())
+      });
+    });
+  });
+  ```
+
+- Something something about how to name your tests
```

Comment:
TODO: How are tests named? I am used to using `it()`/ `describe()`/ `context()` where IMO the test naming makes more sense, not sure exactly what to use as the strings for `suite()` and `test()`.